### PR TITLE
Add AdminBar stories.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #2942 

## Relevant technical choices
* Added `common.stories.js` which contains decorators and functions used by most of the AdminBar components components.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/adminbar/AdminBarClicks.stories.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+import { withWidgetComponentProps } from '../../googlesitekit/widgets/util';
+import { setupSearchConsoleMockReports, widgetDecorators } from './common.stories';
+import AdminBarClicks from './AdminBarClicks';
+import WithRegistrySetup from '../../../../tests/js/WithRegistrySetup';
+
+const WidgetWithComponentProps = withWidgetComponentProps( 'widget-slug' )( AdminBarClicks );
+
+const Template = ( { setupRegistry, ...args } ) => (
+	<WithRegistrySetup func={ setupRegistry ? setupRegistry : () => {} }>
```
Comment (tofumatt) [inline assets/js/components/adminbar/AdminBarClicks.stories.js:30]: This is a bit more concise and I'd argue a bit easier to read—`setupRegistry` appearing twice in a row is a bit tougher to parse to my eyes 🙂 
```suggestion
	<WithRegistrySetup func={ setupRegistry || () => {} }>
```

File: assets/js/components/adminbar/AdminBarImpressions.stories.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+import { withWidgetComponentProps } from '../../googlesitekit/widgets/util';
+import { setupSearchConsoleMockReports, widgetDecorators } from './common.stories';
+import AdminBarImpressions from './AdminBarImpressions';
+import WithRegistrySetup from '../../../../tests/js/WithRegistrySetup';
+
+const WidgetWithComponentProps = withWidgetComponentProps( 'widget-slug' )( AdminBarImpressions );
+
+const Template = ( { setupRegistry, ...args } ) => (
+	<WithRegistrySetup func={ setupRegistry ? setupRegistry : () => {} }>
```
Comment (tofumatt) [inline assets/js/components/adminbar/AdminBarImpressions.stories.js:30]: Same as above/for all instances here.

Also, you could do:
```suggestion
const Template = ( { setupRegistry =  () => {}, ...args } ) => (
	<WithRegistrySetup func={ setupRegistry }>
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
<del>
The stories themselves look good here, but the naming of everything seems a bit redundant. The stories end up looking like this:
</del>
Screenshot 2021-06-09 at 18 21 28
<del>
Which I think is a bit too many instances of "AdminBar".
</del>
(See my second review below, I think I'm wrong about this 😆 )

The "Ready" and "DataUnavailable" are odd names because in the case of "Zero Data" the "Ready" state is the only one used which is a bit odd to parse—I get how that works but I don't think it's immediately intuitive.

Maybe "Loaded" and "Not loaded" as that's what they are, right?

So the overall approach is good but a few coding style tweaks and maybe some improvements to naming so there's less repetition. But if that request is too vague let me know and I can help with some suggestions 😅

**Review by tofumatt — CHANGES_REQUESTED** [review]
Upon reflection, I guess the repetition is because those are all the component names. That's actually helpful then and shouldn't be changed; I'm just not used to seeing it. 🤦🏻 

But maybe we can still adjust the "Ready" and "DataUnavailable" to be more descriptive?

File: assets/js/components/adminbar/common.stories.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		},
+		{
+			slug: 'analytics',
+			active: true,
+			connected: true,
+		},
+	] );
+
+	// Call story-specific setup.
+	if ( args?.setupRegistry && typeof args?.setupRegistry === 'function' ) {
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/adminbar/common.stories.js]: 👍

File: assets/js/components/adminbar/AdminBarZeroData.stories.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Internal dependencies
+ */
+import AdminBarZeroData from './AdminBarZeroData';
+
+const Template = ( args ) => <AdminBarZeroData { ...args } />;
+
+export const Ready = Template.bind( {} );
+Ready.storyName = 'Ready';
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/adminbar/AdminBarZeroData.stories.js]: @tofumatt I've renamed it to "Data Unavailable" to be consistent with how we did with other stories (new v6 ones). Let me know if that sounds good.

File: assets/js/components/adminbar/AdminBarWidgets.stories.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+		( Story, { args } ) => {
+			const setupRegistry = ( registry ) => {
+				// Set some site information.
+				provideSiteInfo( registry, {
+					currentEntityURL: 'https://www.sitekitbygoogle.com/blog/',
+					currentEntityTitle: 'Blog test post for Google Site Kit',
+				} );
+
+				// Call story-specific setup.
+				if ( args?.setupRegistry && typeof args?.setupRegistry === 'function' ) {
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/adminbar/AdminBarWidgets.stories.js]: 👍

File: assets/js/components/adminbar/AdminBarClicks.stories.js (hunk 1 above)
Comment (asvinb) [reply assets/js/components/adminbar/AdminBarClicks.stories.js]: It was tough to parse in my eyes as well, 🤣  however when i tried it, I was getting a syntax error. I went with the other solution where you suggested we set the default parameter value.

File: assets/js/components/adminbar/AdminBarImpressions.stories.js (hunk 2 above)
Comment (asvinb) [reply assets/js/components/adminbar/AdminBarImpressions.stories.js]: 👍

**Comment by asvinb** [comment]
@tofumatt I kept the "Ready" and "Data Unavailable" naming convention so that it's consistent with what we did for other stories.
image

Do you have any suggestions?

**Review by tofumatt — APPROVED** [review]
Looks good 👍🏻 

Thanks for the clarification on the story names, makes sense to keep 'em consistent.

Thanks for the tweaks as well, looks good 👍🏻

**Pull request merged**